pkg/cmd/dcos_cluster_list.go

```diff
@@ -0,0 +1,106 @@
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"sync"
+
+	"github.com/dcos/dcos-cli/pkg/cli"
+	"github.com/dcos/dcos-cli/pkg/config"
+	"github.com/dcos/dcos-cli/pkg/httpclient"
+	"github.com/spf13/cobra"
+)
+
+// newCmdClusterList lists the clusters.
+func newCmdClusterList(ctx *cli.Context) *cobra.Command {
+	var onlyAttached bool
+	var jsonOutput bool
+	cmd := &cobra.Command{
+		Use:  "list",
+		Args: cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			manager := ctx.ConfigManager()
+			confs := manager.All()
+
+			currentConf, err := manager.Current()
+			if err != nil {
+				return err
+			}
+			currentConfID := filepath.Base(filepath.Dir(currentConf.Path()))
+
+			// Clusters is composed of cluster containing its name, id, status, version, and url.
+			clusters := [][]string{}
+
+			var wg sync.WaitGroup
+			for _, conf := range confs {
+				// Add should execute before the statement creating the goroutine.
+				wg.Add(1)
+				go func(conf *config.Config) {
+					defer wg.Done()
+
+					name := conf.Get("cluster.name").(string)
+					id := filepath.Base(filepath.Dir(conf.Path()))
+					if id == currentConfID {
+						name += "*"
+					} else if onlyAttached {
+						return
+					}
+					url := conf.Get("core.dcos_url").(string)
+					status := "AVAILABLE"
+					version, err := getVersion(url)
+					if err != nil {
+						status = "UNAVAILABLE"
+					}
+					clusters = append(clusters, []string{name, id, status, version, url})
```

Comment:
read and writes to `clusters` happen concurrently, it needs to be protected with a mutex.